app/controllers/support_interface/ucas_matches_controller.rb

```diff
@@ -1,7 +1,13 @@
 module SupportInterface
   class UCASMatchesController < SupportInterfaceController
     def index
-      @matches = UCASMatch.includes(:candidate)
+      @filter = SupportInterface::MatchesFilter.new(params: params)
+      @matches = @filter.filter_records(
+        UCASMatch
+        .includes(:candidate)
+        .order(updated_at: :desc)
+        .page(params[:page] || 1).per(15),
```

Comment:
Shouldn't we paginate after filtering? in this case we could get say 15 records then filter out 5?